### PR TITLE
Allow project deployment only via entrypoint

### LINKED ISSUE #9860: Consolidate projects and deployments
The 'project' concept has become unnecessary as the new deployment UX has developed and can be removed to avoid conflation with the 'project' concept from Prefect 1.0. Remove mentions of 'projects' from Prefect 2. This will impact the following areas:

### Functionality

- [x]  Remove creation and use of the `.prefect` folder that is currently created by `prefect project init`

- [x]  Remove the ability to deploy a flow with a flow name and instead use entrypoint universally

### CLI

- [x]  `prefect project init` -> `prefect init`

- [x]  `prefect project recipe ls` -> `prefect init recipes ls`

- [x]  Remove `prefect project clone`

### Modules

- [ ]  `prefect.projects` -> `prefect.deployments`

    - Note that steps would now be referenced `prefect.deployment.steps.step_name`

- [ ]  `prefect_aws.projects` -> `prefect_aws.deployments`

- [ ]  `prefect_gcp.projects` -> `prefect_gcp.deployments`

- [ ]  `prefect_azure.projects` -> `prefect_azure.deployments`

- [ ]  `prefect_docker.projects` -> `prefect_docker.deployments`

### Step Names

- [x]  `prefect`

     `git_clone_project` -> `git_clone`

- [ ]  `prefect-aws`

    - `pull_project_from_s3` -> `pull_from_s3`
    - `push_project_to_s3` -> `push_to_s3`

- [ ]  'prefect-gcp`

    - `pull_project_from_gcs` -> `pull_from_gcs`
    - `push_project_to_gcs` -> `push_to_gcs`

- [ ]  `prefect-azure`

    - `pull_project_from_azure_blob_storage` -> `pull_from_azure_blob_storage`
    - `push_project_to_azure_blob_storage` -> `push_to_azure_blob_storage`

### PR DESCRIPTION
**Pull request by serinamarie**
## Overview
PR 1 to close #9860
(PR 2 will target the actual removal of the word `project` from modules, the docs, and CLI)

This PR targets the section of the issue focused on modifying functionality, meaning:
🗑️ Removing creation and use of the `.prefect` folder that is currently created by `prefect project init`
🚫 Removing the ability to deploy a flow with a flow name and instead use entrypoint universally

**What does that mean for users?**
The good:
✔️ A single way to deploy a flow in projects (specifying entrypoint)
✔️ No more `.prefect` folder created
Caveats:
⚠️ A breaking change: You won't be able to deploy a flow via flow name, even if you have an existing `.prefect` folder with registered flows.

**TODO**
📚 Documentation updates regarding functionality changes

#### `prefect project init`
```bash
❯ cd hello-projects
❯ prefect project init
11:52:52.782 | DEBUG   | prefect.profiles - Using profile 'local_sqlite'
Created project in /Users/code/prefect/hello-projects with the following new files:
.prefectignore
deployment.yaml
prefect.yaml
```
#### `prefect deploy`
👍 Still correct
```bash
❯ prefect deploy flow.py:main -n 'main-deployment'
11:53:31.741 | DEBUG   | prefect.profiles - Using profile 'local_sqlite'
11:53:32.889 | DEBUG   | prefect.client - Using ephemeral application with database at sqlite+aiosqlite:////Users/bean/.prefect/prefect.db
? Would you like to schedule when this flow runs? [y/n] (y): n
...
```

🔔 Raises new error
```bash
❯ prefect deploy -f main
11:55:50.076 | DEBUG   | prefect.profiles - Using profile 'local_sqlite'
Usage: prefect deploy [OPTIONS] [ENTRYPOINT]
Try 'prefect deploy --help' for help.
╭─ Error ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ No such option: -f    
```

🔔 Raises helpful error now instead of "not enough values to unpack (expected 2, got 1)"
```bash
❯ prefect deploy main
12:01:14.694 | DEBUG   | prefect.profiles - Using profile 'local_sqlite'
12:01:15.892 | DEBUG   | prefect.client - Using ephemeral application with database at sqlite+aiosqlite:////Users/bean/.prefect/prefect.db
Entrypoint 'main' is not in the format `path/to/file.py:flow_function`
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: src/prefect/cli/deploy.py [hunk 1]
```diff
@@ -425,7 +377,7 @@ async def _run_single_deploy(
 
     ## parse parameters
     # minor optimization in case we already loaded the flow
-    if not flow:
+    if not flow:  # TK: not sure if we need this anymore?
```
Comment (serinamarie) [inline src/prefect/cli/deploy.py:380]: I was unsure if there are still situations where we need this

File: src/prefect/cli/deploy.py [hunk 2]
```diff
@@ -355,67 +347,27 @@ async def _run_single_deploy(
...
+    except ModuleNotFoundError:
+        raise ValueError(
+            f"Could not find a flow at {entrypoint}.\n\nPlease ensure your"
+            " entrypoint is in the format `path/to/file.py:flow_fn_name` and the"
+            " file name and flow function name are correct."
+        )
+    flow_name = flow.name  # TK: not sure if should keep
 
-    base_deploy["flow_name"] = flow_name
+    base_deploy["flow_name"] = flow_name  # TK: not sure if should keep?
```
Comment (serinamarie) [inline src/prefect/cli/deploy.py:370]: I am unsure if we want to keep this

**Comment by madkinsz** [comment]
Would it make sense to move all of the `.prefect` loading logic into the `deprecated` namespace and still perform the old behavior if the directory exists but throw deprecation warnings? Breaking changes are not ideal.

**Comment by serinamarie** [comment]
> Would it make sense to move all of the `.prefect` loading logic into the `deprecated` namespace and still perform the old behavior if the directory exists but throw deprecation warnings? Breaking changes are not ideal.

We had decided it would be a breaking change, but I'll defer to @desertaxle.

**Comment by desertaxle** [comment]
After considering it, I agree we should deprecate the `--flow-name` flag instead of removing it entirely. That will allow us to guide any users deploying via flow name to use the entrypoint option. I'd like to remove the`-f` and `.prefect` functionality when the new deployment UX exits beta.

**Comment by serinamarie** [comment]
> After considering it, I agree we should deprecate the `--flow-name` flag instead of removing it entirely. That will allow us to guide any users deploying via flow name to use the entrypoint option. I'd like to remove the`-f` and `.prefect` functionality when the new deployment UX exits beta.

So that I'm understanding correctly, deploying via flow name will end up deprecated for just a few weeks (when deployment UX exits beta) before removal?

**Comment by serinamarie** [comment]
We will wait until we're ready for deployments to exit beta.

**Comment by serinamarie** [comment]
We have decided to extend the deprecation period to six months. So I am closing this until December (do not open until Christmas).

**Pull request closed without merging**